### ISSUE TITLE
Block depth below 0 error in version 6.3.0

### ISSUE THREAD

**Issue by jamesball-inexus** [flyway/flyway#2720]
###### Which version and edition of Flyway are you using?

6.3.0

###### If this is not the latest version, can you reproduce the issue with the latest one as well?

n/a

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)

Gradle plugin

###### Which database are you using (type & version)?

MySQL 8.0.19

###### Which operating system are you using?

Windows 10 (MySQL running in Linux Docker container)

###### What did you do?

Hi,

I have found the below migration crashes due to the following error: `Flyway parsing bug: unable to decrease block depth below 0`.

SQL with schema, table, variable names removed:

```
DROP EVENT [..].[..];

DELIMITER //
CREATE EVENT [..].[..]
ON SCHEDULE EVERY 1 SECOND STARTS CURRENT_TIMESTAMP()
ON COMPLETION PRESERVE
ENABLE
DO
[...]: BEGIN
    DECLARE [...] SMALLINT;

    DECLARE [...] DATE;

    SELECT
        s.[...],
        s.[...]
    FROM
        [...].[...] AS s
    WHERE
        s.[...] = 1
    INTO
        [...],
        [...];

    /*
     *  [...]
     */
    IF NOT EXISTS (
        SELECT
            NULL
        FROM
            [...].[...] AS rh
        WHERE
            rh.[...] = [...]
    ) THEN
        CALL [...].[...]([...], [...]);
    END IF;
END [...]//
DELIMITER ;
```

###### What did you expect to see?

THe migration executed successfully.

###### What did you see instead?

The migration was not executed due to the following error: `Flyway parsing bug: unable to decrease block depth below 0`.

**Comment by MikielAgutu** [comment]
Can you see if this is [fixed in 6.3.1](https://flywaydb.org/documentation/releaseNotes#6.3.1)?

**Comment by jamesball-inexus** [comment]
I have tested in 6.3.1 and the issue is still occurring.

**Comment by juliahayward** [comment]
Will be fixed in 6.3.2
